source/_UIAHandler.py

```diff
@@ -182,11 +182,16 @@ def MTAThreadFunc(self):
 			except (COMError,WindowsError,NameError):
 				self.clientObject=CoCreateInstance(CUIAutomation._reg_clsid_,interface=IUIAutomation,clsctx=CLSCTX_INPROC_SERVER)
 			if isUIA8:
-				try:
-					self.clientObject=self.clientObject.QueryInterface(IUIAutomation3)
-				except COMError:
-					self.clientObject=self.clientObject.QueryInterface(IUIAutomation2)
-			log.info("UIAutomation: %s"%self.clientObject.__class__.__mro__[1].__name__)
+				# #8009: use appropriate interface based on highest supported interface.
+				for interface in (IUIAutomation5, IUIAutomation4, IUIAutomation3, IUIAutomation2):
+					try:
+						self.clientObject=self.clientObject.QueryInterface(interface)
+						break
+					except COMError:
+						pass

```

Comment:
Perhaps an else on the for loop that raises RuntimeError("No UIAutomation client interface supported")